packages/react-composites/src/composites/ChatComposite/file-sharing/FileUpload.ts

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { EventEmitter } from 'events';
+import { nanoid } from 'nanoid';
+
+/**
+ * Meta Data containing information about the uploaded file.
+ * @beta
+ */
+export interface FileMetadata {
+  /**
+   * File name to be displayed.
+   */
+  name: string;
+  /**
+   * Extension is used for rendering the file icon.
+   * An unknown extension will be rendered as a generic icon.
+   * Example: `jpeg`
+   */
+  extension: string;
+  /**
+   * Download URL for the file.
+   */
+  url: string;
+}
+
+/**
+ * @beta
+ */
+export interface FileUploadState {
+  /**
+   * Unique identifier for the file upload.
+   */
+  id: string;
+
+  /**
+   * Filename extracted from the {@link File} object.
+   */
+  filename: string;
+
+  /**
+   * A number between 0 and 1 indicating the progress of the upload.
+   */
+  progress: number;
+
+  /**
+   * Meta Data {@link FileMetadata} containing information about the uploaded file.
+   */
+  metadata?: FileMetadata;
+
+  /**
+   * Error message to be displayed to the user if the upload fails.
+   */
+  errorMessage?: string;
+}
+
+/**
+ * A wrapper object for a file that is being uploaded.
+ * Allows mmanaging file uploads by providing common functions for updating the
+ * upload progress, canceling an upload, completing an upload etc.
+ * @beta
+ */
+export interface FileUploadManager {
+  /**
+   * HTML {@link File} object for the uploaded file.
+   */
+  file: File;
+  /**
+   * Update the progress of the upload.
+   * Emits the `uploadProgressed` event.
```

Comment:
Don't reference the `uploadProgressed` and other events in this interface at all. It's an implementation detail of `class FileUpload` that this triggers the said event.

For Contoso, the more we can do to keep the `FileUploadManager` and `ObservableFileUpload` interfaces separate, the better. (So audit any references across these two interfaces and remove them).